### PR TITLE
unified latency metric unit

### PR DESCRIPTION
**Pull request by StevenLuMT**
Descriptions of the changes in this PR:

### Motivation

The latency of the OpStatsLogger.registerSuccessfulEvent calculation is to convert the time to milliseconds
image
but Counter.add nothing to do

so when using Counter for latency statistics, the time unit and OpStatsLogger are not unified, which is easy to be misleading.
then we unified latency metric unit 

### Changes

1.  change name : Counter.add -->  Counter.addCount
2. add new method Counter.addLatency to count the time and convert the time to milliseconds

then how to use counter correctly:
1. when using Counter for latency metric, call Counter.addLatency
2. when using Counter for count metric, call Counter.addCount

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
Lgtm

But we can't cherry pick this into released branches

**Comment by Shoothzj** [comment]
will this break prometheus metric result?

**Comment by StevenLuMT** [comment]
> will this break prometheus metric result?

this change don't break prometheus metric result:
This just optimize the statistics,but in the end this logic still use Counter.add to update the data
@Shoothzj 
image

**Review by Shoothzj — APPROVED** [review]

**Review by wenbingshen — APPROVED** [review]
nice found.

**Review by dlg99 — APPROVED** [review]

File: bookkeeper-common/src/test/java/org/apache/bookkeeper/test/TestStatsProvider.java [hunk 1]
```diff
@@ -60,10 +60,16 @@ public void dec() {
...
 
         @Override
-        public void add(long delta) {
+        public void addCount(long delta) {
             updateMax(val.addAndGet(delta));
         }
 
+        @Override
+        public void addLatency(long eventLatency, TimeUnit unit) {
+            long valueMillis = unit.toMicros(eventLatency) / 1000;
```
Comment (zymap) [inline bookkeeper-common/src/test/java/org/apache/bookkeeper/test/TestStatsProvider.java:69]: Just have a question. Why not use unit.toMills?

Comment (StevenLuMT) [reply bookkeeper-common/src/test/java/org/apache/bookkeeper/test/TestStatsProvider.java:69]: > Just have a question. Why not use unit.toMills?

it's the same, just use the logic with OpStatsLogger.registerSuccessfulEvent @zymap 
image

Comment (Shoothzj) [reply bookkeeper-common/src/test/java/org/apache/bookkeeper/test/TestStatsProvider.java:69]: I think we can both use unit.toMills

**Review by zymap — APPROVED** [review]

File: bookkeeper-common/src/test/java/org/apache/bookkeeper/test/TestStatsProvider.java (hunk 1 above)
Comment (StevenLuMT) [reply bookkeeper-common/src/test/java/org/apache/bookkeeper/test/TestStatsProvider.java:69]: > I think we can both use unit.toMills

goog idea, I will update it

**Comment by StevenLuMT** [comment]
@Shoothzj  have a look the new update for unit.toMillis

File: bookkeeper-common/src/test/java/org/apache/bookkeeper/test/TestStatsProvider.java (hunk 1 above)
Comment (StevenLuMT) [reply bookkeeper-common/src/test/java/org/apache/bookkeeper/test/TestStatsProvider.java]: @Shoothzj have a look the new update for unit.toMillis

**Comment by StevenLuMT** [comment]
@Shoothzj have a look the new update for unit.toMillis

**Comment by eolivelli** [comment]
@Shoothzj this is a PR related to a BP and we did not run the VOTE.
I will send a message on dev@

We should follow the community process

**Comment by Shoothzj** [comment]
@eolivelli Sorry, My fault, I didn't realize it. Should we revert the PR now or wait for the vote result?

**Comment by StevenLuMT** [comment]
> @eolivelli Sorry, My fault, I didn't realize it. Should we revert the PR now or wait for the vote result?

I have started the VOTE, see for details https://lists.apache.org/thread/4kxbdwjmls56lfmfhrf853dc525dmsy9
@Shoothzj @eolivelli @dlg99 @zymap

**Comment by StevenLuMT** [comment]
> @eolivelli Sorry, My fault, I didn't realize it. Should we revert the PR now or wait for the vote result?

@Shoothzj don't revert ,see the detail for mail discussion:  
image

https://lists.apache.org/thread/zyj3z3tt4bsx928mc0ogfkv63w9t47bc

**Pull request merged**